src/script/lua_api/l_camera.cpp

```diff
@@ -115,38 +120,40 @@ int LuaCamera::l_get_offset(lua_State *L)
 	return 1;
 }
 
+// get_look_dir(self)
 int LuaCamera::l_get_look_dir(lua_State *L)
 {
-	LocalPlayer *player = getClient(L)->getEnv().getLocalPlayer();
-	sanity_check(player);
-
-	float pitch = -1.0 * player->getPitch() * core::DEGTORAD;
-	float yaw = (player->getYaw() + 90.) * core::DEGTORAD;
-	v3f v(std::cos(pitch) * std::cos(yaw), std::sin(pitch),
-			std::cos(pitch) * std::sin(yaw));
+	Camera *camera = getobject(L, 1);
+	if (!camera)
+		return 0;
 
-	push_v3f(L, v);
+	push_v3f(L, camera->getDirection());
 	return 1;
 }
 
+// get_look_horizontal(self)
+// FIXME: wouldn't localplayer be a better place for this?
```

Comment:
Would be consistent with the server-side `ObjectRef`, but since these values are calculated locally I think they should be kept in the camera object.